elki-outlier/src/main/java/elki/outlier/isolation/IsolationForest.java

```diff
@@ -0,0 +1,366 @@
+import elki.result.outlier.BasicOutlierScoreMeta;
+import elki.result.outlier.OutlierResult;
+import elki.result.outlier.OutlierScoreMeta;
+import elki.utilities.Alias;
+import elki.utilities.documentation.Description;
+import elki.utilities.documentation.Reference;
+import elki.utilities.documentation.Title;
+import elki.utilities.optionhandling.OptionID;
+import elki.utilities.optionhandling.Parameterizer;
+import elki.utilities.optionhandling.constraints.CommonConstraints;
+import elki.utilities.optionhandling.parameterization.Parameterization;
+import elki.utilities.optionhandling.parameters.IntParameter;
+import elki.utilities.optionhandling.parameters.RandomParameter;
+import elki.utilities.random.RandomFactory;
+
+import net.jafama.FastMath;
+
+/**
+ * Isolation-Based Anomaly Detection.
+ * <p>
+ * This algorithm detects anomalies purely based on the concept of
+ * isolation without employing any distance or density measure.
+ * <p>
+ * Reference:
+ * <p>
+ * Fei Tony Liu, Kai Ming Ting, Zhi-Hua Zhou<br>
+ * Isolation-Based Anomaly Detection<br>
+ * ACM Transanctions on Knowledge Discovery from Data (TKDD'12)
+ * 
+ * @author Fei Tony Liu et al. (Original Code)
+ * @author Braulio V.S. Vinces (ELKIfication)
+ *
+ * @param <V> Vector type
+ */
+@Title("IsolationForest: Isolation-Based Anomaly Detection")
+@Description("This algorithm detects anomalies purely based on the concept of isolation without employing any distance or density measure.")
+@Reference(authors = "Fei Tony Liu, Kai Ming Ting, Zhi-Hua Zhou", //
+    title = "Isolation-Based Anomaly Detection", //
+    booktitle = "ACM Transanctions on Knowledge Discovery from Data (TKDD'12)", //
+    url = "https://doi.org/10.1145/2133360.2133363", //
+    bibkey = "DBLP:journals/tkdd/LiuTZ12")
+@Alias({ "de.lmu.ifi.dbs.elki.algorithm.outlier.IsolationForest", "iforest" })
+public class IsolationForest implements OutlierAlgorithm {
+
+  /**
+   * Class logger
+   */
+  private static final Logging LOG = Logging.getLogger(IsolationForest.class);
+
+  /*
+   * The set of trees
+   */
+  protected List<Tree> trees = null;
+
+  /*
+   * The number of trees
+   */
+  protected int numTrees = 100;
+
+  /*
+   * The sub sample size
+   */
+  protected int subsampleSize = 256;
+
+  /**
+   * Random generator
+   */
+  private RandomFactory rnd;
+
+  /**
+   * Constructor.
+   *
+   * @param numTrees
+   * @param subSamplingSize
+   * @param rnd
+   */
+  public IsolationForest(int numTrees, int subSamplingSize, RandomFactory rnd) {
+    this.numTrees = numTrees;
+    this.subsampleSize = subSamplingSize;
+    this.rnd = rnd;
+  }
+
+  public OutlierResult run(Database db, Relation<? extends NumberVector> relation) {
+    // Output data storage
+    WritableDoubleDataStore scores = DataStoreUtil.makeDoubleStorage(relation.getDBIDs(), DataStoreFactory.HINT_DB);
+    FiniteProgress prog = LOG.isVerbose() ? new FiniteProgress("IsolationForest progress", relation.size(), LOG) : null;
```

Comment:
The `prog` progress should run only until numTrees, not `relation.size()`, and you'll likely want a second progress for scoring.